ext/gmp/gmp.c

```diff
@@ -480,7 +480,8 @@ static int gmp_compare(zval *op1, zval *op2) /* {{{ */
 		return ZEND_UNCOMPARABLE;
 	}
 
-	return ZEND_THREEWAY_COMPARE(mpz_cmp(gmp_op1, gmp_op2), 0);
+	int ret = mpz_cmp(gmp_op1, gmp_op2); /* avoid multiple evaluations */
+	return ZEND_THREEWAY_COMPARE(ret, 0);
```

Comment:
is it somehow possible to put this into the macro itself?

Reply:
Not without relying on non-portable GCC extensions.